app/scripts/controllers/edit/buildConfig.js

```diff
@@ -348,6 +353,36 @@ angular.module('openshiftConsole')
       return triggers;
     };
 
+    var loadBuildConfigSecrets = function() {
+      $scope.secrets.picked = {
+        gitSecret: $scope.buildConfig.spec.source.sourceSecret || {name: ""},
+        pullSecret: buildStrategy($scope.buildConfig).pullSecret || {name: ""},
+        pushSecret: $scope.buildConfig.spec.output.pushSecret || {name: ""},
+      }
+
+      if ($scope.strategyType === "Source" || $scope.strategyType === "Docker") {
+        $scope.secrets.picked.sourceSecrets = $scope.buildConfig.spec.source.secrets || [{secret: { name: ""}, destinationDir: ""}];
+      } else if ($scope.strategyType === "Custom") {
+        $scope.secrets.picked.sourceSecrets = buildStrategy($scope.buildConfig).secrets || [{secretSource: { name: ""}, mountPath: ""}];
+      }
+    }
+
+    var updateSecrets = function(object, pickedSecret, secretFieldName) {
+      if (!_.isEmpty(pickedSecret.name)) {
+        object[secretFieldName] = pickedSecret;
+      } else {
+        delete object[secretFieldName];
+      }
+    }
+
+    var updateSourceSecrets = function(object, pickedSecrets) {
+      if (!_.isEmpty(pickedSecrets[0][$scope.strategyType === "secretSource" ? "mountPath" : "secret"].name)) {
```

Comment:
Maybe

```js
var pickedSecret = _.head(pickedSecrets);
var property = $scope.strategyType === "secretSource" ? "mountPath" : "secret";
if (_.has(pickedSecret, [property, "name"]) ...
```

Although I'm confused by this. Can you explain why we're checking mountPath.name in some cases?